src/mailkit.cs

```diff
@@ -0,0 +1,423 @@
+using System;
+using CoreGraphics;
+using Foundation;
+using ObjCRuntime;
+using AVFoundation;
+using AppKit;
+
+namespace MailKit {
+	[NoWatch, NoTV, NoiOS, NoMacCatalyst, Mac (12,0)]
+	[Native, Advice ("'MEComposeSessionErrorCode' is not available in UIKit on macOS.")]
+	[ErrorDomain ("MEComposeSessionErrorDomain")]
+	public enum MEComposeSessionErrorCode : long
+	{
+		Recipients = 0,
+		Headers = 1,
+		Body = 2,
+	}
+
+	[NoWatch, NoTV, NoiOS, NoMacCatalyst, Mac (12,0)]
+	[Native, Advice ("'MEMessageActionMessageColor' is not available in UIKit on macOS.")]
+	public enum MEMessageActionMessageColor : long
+	{
+		None,
+		Green,
+		Yellow,
+		Orange,
+		Red,
+		Purple,
+		Blue,
+		Gray,
+	}
+
+	[NoWatch, NoTV, NoiOS, NoMacCatalyst, Mac (12,0)]
+	[Native, Advice ("'MEMessageState' is not available in UIKit on macOS.")]
+	public enum MEMessageState : long
+	{
+		Received = 0,
+		Draft = 1,
+		Sending = 2,
+	}
+
+	[NoWatch, NoTV, NoiOS, NoMacCatalyst, Mac (12,0)]
+	[Advice ("'MEMessage' is not available in UIKit on macOS.")]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface MEMessage : NSSecureCoding
+	{
+		[Export ("state", ArgumentSemantic.Assign)]
+		MEMessageState State { get; }
+
+		[Export ("subject")]
+		string Subject { get; }
+
+		[Export ("fromAddress", ArgumentSemantic.Copy)]
+		MEEmailAddress FromAddress { get; }
+
+		[Export ("toAddresses", ArgumentSemantic.Copy)]
+		MEEmailAddress[] ToAddresses { get; }
+
+		[Export ("ccAddresses", ArgumentSemantic.Copy)]
+		MEEmailAddress[] CcAddresses { get; }
+
+		[Export ("bccAddresses", ArgumentSemantic.Copy)]
+		MEEmailAddress[] BccAddresses { get; }
+
+		[Export ("replyToAddresses", ArgumentSemantic.Copy)]
+		MEEmailAddress[] ReplyToAddresses { get; }
+
+		[Export ("allRecipientAddresses", ArgumentSemantic.Copy)]
+		MEEmailAddress[] AllRecipientAddresses { get; }
+
+		[Export ("dateSent", ArgumentSemantic.Copy)]
+		[NullAllowed]
+		NSDate DateSent { get; }
+
+		[Export ("dateReceived", ArgumentSemantic.Copy)]
+		[NullAllowed]
+		NSDate DateReceived { get; }
+
+		[NullAllowed, Export ("headers", ArgumentSemantic.Copy)]
+		NSDictionary<NSString, NSArray<NSString>> Headers { get; }
```

Comment:
Can this be a smart dictionary ?

Reply:
Sorry, what do you mean by smart dictionary?

Reply:
Sorry, strong dictionary: https://docs.microsoft.com/en-us/xamarin/cross-platform/macios/binding/binding-types-reference#strongly-typed-dictionaries

Reply:
It might be the case it can't since headers can be anything in the mail protocol, so lets ignore it for now